tools/misc/tpm2_checkquote.c

```diff
@@ -160,57 +161,124 @@ static TPM2B_ATTEST *message_from_file(const char *msg_file_path) {
     return msg;
 }
 
-static bool pcrs_from_file(const char *pcr_file_path,
-        TPML_PCR_SELECTION *pcr_select, tpm2_pcrs *pcrs) {
-
-    bool result = false;
-    unsigned long size;
+static bool parse_selection_data_from_selection_string(FILE *pcr_input,
+    TPML_PCR_SELECTION *pcr_select, tpm2_pcrs *pcrs) {
 
-    if (!files_get_file_size_path(pcr_file_path, &size)) {
+    bool result = pcr_parse_selections(ctx.pcr_selection_string, pcr_select);
+    if (!result) {
+        LOG_ERR("Could not parse PCR selections");
         return false;
     }
 
-    if (!size) {
-        LOG_ERR("The pcr file \"%s\" is empty", pcr_file_path);
-        return false;
+    /*
+     * A tpm2_pcrs->pcr_values[tpm2_pcrs->count] is a TPML_DIGEST structure
+     * which can hold a maximum of 8 digests. Once the count of 8 is exhausted
+     * we need a new TPML_DIGEST structure.
+     *
+     * The digests count in a list is tracked with
+     * tpm2_pcrs->pcr_values[tpm2_pcrs->count].count
+     *
+     * A total of such lists is tracked by the tpm2_pcrs->count.
+     */
+    unsigned i = 0;
+    unsigned j = 0;
+    unsigned read_size = 0;
+    size_t read_count = 0;
+    unsigned digest_list_count = 0;
+    for (i = 0; i < pcr_select->count; i++) {
+        read_size = tpm2_alg_util_get_hash_size(pcr_select->pcrSelections[i].hash);
+
+        for (j = 0; j < pcr_select->pcrSelections[i].sizeofSelect * 8; j++) {
+            if ((pcr_select->pcrSelections[i].pcrSelect[j / 8] & 1 << (j % 8)) != 0) {
+                pcrs->pcr_values[digest_list_count].digests[pcrs->pcr_values[digest_list_count].count].size = read_size;
```

Comment:
I **think** scan-build doesn't know that digest_list_count will never be grater than pcr_select->count, so you probably want to check that and make sure of that case.